src/sql/workbench/services/notebook/common/sqlSessionManager.ts

```diff
@@ -227,6 +250,25 @@ class SqlKernel extends Disposable implements nb.IKernel {
 		return this._future;
 	}
 
+	private getCodeWithoutCellMagic(content: nb.IExecuteRequest): string {
+		let code = content.code;
+		let firstLineEnd = code.indexOf(os.EOL);
+		let firstLine = code.substring(0, (firstLineEnd >= 0) ? firstLineEnd : 0).trimLeft();
+		if (firstLine.startsWith('%%')) {
+			// Strip out the line
+			code = code.substring(firstLineEnd, code.length);
+			// Try and match to an external script magic. If we add more magics later, should handle transforms better
+			let magic = notebookUtils.tryMatchCellMagic(firstLine);
+			if (magic) {
+				let executor = this._magicToExecutorMap.get(magic.toLowerCase());
+				if (executor) {
+					code = executor.convertToExternalScript(code);
+				}
+			}
```

Comment:
If we don't find it should the user be notified? how does jupyter handle situations like this? is it just a runtime error?

Reply:
Runtime error